src/main/java/com/google/step/youtube/DescriptionParser.java

```diff
@@ -99,35 +99,50 @@ public Pattern getPattern() {
         return matches;
     }
 
+    /*
+     * Remove duplicate OfferSnippets from originalOffers. Note that the order of elements may change.
+     */
     private static List<OfferSnippet> removeDuplicateOffers(List<OfferSnippet> originalOffers) {
         List<OfferSnippet> noDupsOffers = new ArrayList<>();
         Set<OfferSnippet> offerSet = new HashSet<>(originalOffers);
         noDupsOffers.addAll(offerSet);
-
         return noDupsOffers;
     }
 
     /*
-     * Finds the snippet (line) of description which conatins the target index. 
+     * Finds the snippet of description which conatins the target index. 
      * Bounds the snippet at MAX_SNIPPET_LENGTH characters and does not truncate words.
      */
     private static String getBoundedSnippet(int targetIndex, String description) {
         int startDelimiter = description.lastIndexOf(DELIMITER, targetIndex);
         int endDelimiter = description.indexOf(DELIMITER, targetIndex);
 
         // add 1 to not include delimiter in snippet
-        int snippetStart = startDelimiter == -1 ? 0 : startDelimiter + 1;
-        int snippetEnd = endDelimiter == -1 ? description.length() : endDelimiter;
-        String completeSnippet = description.substring(snippetStart, snippetEnd);
-
-        if (completeSnippet.length() <= MAX_SNIPPET_LENGTH) {
-            return completeSnippet;
+        int lineStart = startDelimiter < 0 ? 0 : startDelimiter + 1;
+        int lineEnd = endDelimiter < 0 ? description.length() : endDelimiter;
+
+        int startBoundIndex = targetIndex - (MAX_SNIPPET_LENGTH / 2);
+        int endBoundIndex = targetIndex + (MAX_SNIPPET_LENGTH / 2);
+        // could factor out description.lastIndexOf(" ", endBoundIndex) and indexOf
```

Comment:
could factor out description.lastIndexOf(" ", endBoundIndex) and indexOf into he

Reply:
whoops, removed comment